### PR TITLE
Refactor / Require all prerequisites needed in the Sign Message controller and improve the controller tests

### PR DESCRIPTION
**Pull request by superKalo**
Covers corner cases for the Sing Message controller:

- Fixed: Require all prerequisites when initiating the controller (Accounts controller to have been loaded) and when signing (to have pulled at least once an account state for the network on which a sign message request gets received).

Also, improves a bit the Sign Message tests by:

- Change: Initiate a real `AccountsController` (instead of a dummy one)
- Change: Migrate the pattern for writing tests to the one proposed here: https://github.com/AmbireTech/ambire-app/issues/2454

### CODE REVIEW

File: src/controllers/signMessage/signMessage.test.ts [hunk 1]
```diff
@@ -37,106 +37,74 @@ const account: Account = {
...
-    } as AccountsController
+    const accountsCtrl = new AccountsController(
+      storage,
+      providersCtrl,
+      networksCtrl,
+      () => {},
+      () => {}
+    )
+    // Internally, it waits Networks and Providers to be loaded
+    await accountsCtrl.initialLoadPromise
```
Comment (sonytooo) [inline src/controllers/signMessage/signMessage.test.ts:78]: The SignMessageController should wait for accounts to be initialized and ready in most of its methods. You should not wait for that in the test. Otherwise, you test only the happy case when everything is loaded

File: src/controllers/signMessage/signMessage.test.ts [hunk 2]
```diff
@@ -37,106 +37,74 @@ const account: Account = {
...
+
+    await accountsCtrl.updateAccountState(account.addr, 'latest')
```
Comment (sonytooo) [inline src/controllers/signMessage/signMessage.test.ts:80]: same here: do not "await"

File: src/controllers/signMessage/signMessage.test.ts (hunk 1 above)
Comment (superKalo) [reply src/controllers/signMessage/signMessage.test.ts]: Yes, I agree. Changed in https://github.com/AmbireTech/ambire-common/pull/880/commits/e24c57df77b0662b9d4d7e45a0348c251917f47a and https://github.com/AmbireTech/ambire-common/pull/880/commits/b754f3c4ec1440482639b57858004979e40f11a1

File: src/controllers/signMessage/signMessage.test.ts (hunk 2 above)
Comment (superKalo) [reply src/controllers/signMessage/signMessage.test.ts]: Yes, no longer needed after the changes I mentioned in https://github.com/AmbireTech/ambire-common/pull/880#discussion_r1679079120

File: src/controllers/signMessage/signMessage.ts [hunk 3]
```diff
@@ -180,7 +188,13 @@ export class SignMessageController extends EventEmitter {
         throw new Error('Network not supported on Ambire. Please contract support.')
       }
 
-      const accountState = this.#accounts.accountStates[account.addr][network!.id]
+      // Could (rarely) happen if not even a single account state is fetched yet
+      const shouldForceUpdateAndWaitForAccountState =
+        !this.#accounts.accountStates[account.addr]?.[network.id]
+      if (shouldForceUpdateAndWaitForAccountState)
+        await this.#accounts.updateAccountState(account.addr, 'latest', [network.id])
```
Comment (sonytooo) [inline src/controllers/signMessage/signMessage.ts:195]: anti-pattern

Comment (sonytooo) [reply src/controllers/signMessage/signMessage.ts:195]: only the main ctrl should call methods of another sub-ctrl to update the state

**Comment by superKalo** [comment]
Simo said in an offline discussion, approved! :)

File: src/controllers/main/main.ts [hunk 4]
```diff
@@ -444,6 +444,15 @@ export class MainController extends EventEmitter {
...
 
   async handleSignMessage() {
+    const accountAddr = this.signMessage.messageToSign?.accountAddr
+    const networkId = this.signMessage.messageToSign?.networkId
+
+    // Could (rarely) happen if not even a single account state is fetched yet
+    const shouldForceUpdateAndWaitForAccountState =
+      accountAddr && networkId && !this.accounts.accountStates[accountAddr]?.[networkId]
+    if (shouldForceUpdateAndWaitForAccountState)
+      await this.accounts.updateAccountState(accountAddr, 'latest', [networkId])
```
Comment (Ivshti) [inline src/controllers/main/main.ts:454]: should we blow up if even after this, the account state is not there?

**Review by Ivshti — COMMENTED** [review]
question

File: src/controllers/main/main.ts (hunk 4 above)
Comment (superKalo) [reply src/controllers/main/main.ts:454]: Yes, I think so. Improved in https://github.com/AmbireTech/ambire-common/pull/885
